### PR TITLE
add canon driver for mf printer/fax/scanner

### PR DESCRIPTION
**Pull request by tie624**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask-drivers/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask-drivers/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

File: Casks/canon-fax.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+cask "canon-fax" do
+  version "4.15.6"
+  sha256 "14d274075bd9818525c2ceb72f64b412731a058f2832740f5dd1a33bb838da58"
+
+  # gdlp01.c-wss.com/gds/ was verified as official when first introduced to the cask
+  url "https://gdlp01.c-wss.com/gds/3/0100010683/04/mac-fax-v#{version.delete(".")}-01.dmg"
+  name "Canon Fax Driver & Utilities"
+  desc "Fax Driver & Utilities for Canon MF Printer"
```
Comment (miccal) [inline Casks/canon-fax.rb:8]:
```suggestion
  desc "Fax driver & utilities for Canon MF printers"
```

Comment (miccal) [reply Casks/canon-fax.rb:8]: For which model is this for?

File: Casks/canon-fax.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+cask "canon-fax" do
+  version "4.15.6"
+  sha256 "14d274075bd9818525c2ceb72f64b412731a058f2832740f5dd1a33bb838da58"
+
+  # gdlp01.c-wss.com/gds/ was verified as official when first introduced to the cask
+  url "https://gdlp01.c-wss.com/gds/3/0100010683/04/mac-fax-v#{version.delete(".")}-01.dmg"
```
Comment (miccal) [inline Casks/canon-fax.rb:6]:
```suggestion
  url "https://gdlp01.c-wss.com/gds/3/0100010683/04/mac-fax-v#{version.no_dots}-01.dmg"
```

File: Casks/canon-fax.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  name "Canon Fax Driver & Utilities"
+  desc "Fax Driver & Utilities for Canon MF Printer"
+  homepage "https://www.usa.canon.com/internet/portal/us/home/support/drivers-downloads"
+
+  depends_on macos: ">= :yosemite"
+
+  pkg "Canon_FAX_Installer.pkg"
+
+  uninstall pkgutil: "jp.co.canon.CUPSFAX.+"
```
Comment (miccal) [inline Casks/canon-fax.rb:15]: Why `+` and not `*`?

File: Casks/canon-fax.rb (hunk 1 above)
Comment (tie624) [reply Casks/canon-fax.rb:8]: I personally have the Canon ImageClass MF634Cdw. But the drivers should work for all their ImageClass MF printers.

File: Casks/canon-fax.rb (hunk 3 above)
Comment (tie624) [reply Casks/canon-fax.rb:15]: I can change it to `*`. I used `+` because all the IDs had something after `CUPSFAX`.

**Comment by tie624** [comment]
I have made the suggested changes.

**Comment by miccal** [comment]
Thank you for your contribution to `homebrew-cask-drivers` @tie624.

**Review by miccal — APPROVED** [review]

**Pull request merged**